src/poetry/console/commands/version.py

```diff
@@ -8,14 +8,14 @@
 from poetry.core.version.exceptions import InvalidVersion
 from tomlkit.toml_document import TOMLDocument
 
-from poetry.console.commands.command import Command
+from poetry.console.commands.installer_command import InstallerCommand
 
 
 if TYPE_CHECKING:
     from poetry.core.constraints.version import Version
 
 
-class VersionCommand(Command):
+class VersionCommand(InstallerCommand):
```

Comment:
I don't like that. `InstallerCommand` requires the creation of a virtualenv. We shouldn't make commands `InstallerCommand` if it's not required. Is touching the lock file really required here to cover your use case? (Bumping the version does not change the lock file.)

Reply:
This feature is about keeping `poetry.lock`'s modification time newer than `pyproject.toml` so that external tooling like Make is able to discern whether or not the lockfile is out of date. https://github.com/python-poetry/poetry/issues/7392 explains the use case.

Even though the lockfile contents aren't touched here, pyproject.toml is, meaning that Make would think the lockfile is out of date.

I suppose `VersionCommand` could alternatively write to `pyproject.toml` without changing its modificaiton time (perhaps by resetting it)?

Reply:
> I suppose `VersionCommand` could alternatively write to `pyproject.toml` without changing its modificaiton time (perhaps by resetting it)?

That seems also wrong to me. `VersionCommand` changes pyproject.toml and thus should change the modification date of pyproject.toml. However, it doesn't change the lock file and thus shouldn't change the modification date of the lock file. Adding quirks to suffice the `make` use case doesn't seem very attractable to me.

Reply:
I'm up for only applying this to `add` and `remove` -- I'll make the change.

Any thoughts on testing? This [section in `commands/add.py`](https://github.com/python-poetry/poetry/blob/85ef11bc6f0cc0dad9c630351688ae4b98ea34be/src/poetry/console/commands/add.py#L244-L248) makes it "impossible" for `TestLocker` to have `self._write == True` (to materialize test lockfile to disk & direct test modification time).